src/entity/Car.java

```diff
@@ -0,0 +1,51 @@
+package entity;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.logging.Logger;
+
+public class Car {
+    private int id;
+    private AtomicBoolean free = new AtomicBoolean(true);
+    private ReentrantLock lock;
+
+    public static final Logger logger = Logger.getLogger(Car.class.getName());
+
+
+    public Car(int id, ReentrantLock lock) {
+        this.id = id;
+        this.lock = lock;
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public AtomicBoolean isFree() {
+        return free;
+    }
+
+    public void occupy() {
+        lock.lock();
```

Comment:
If you will do the logic to receive order for a company inside it's method, then try to think how and when does you have to use lock.